pubkey_encrypt_phpseclib/src/Plugin/AsymmetricKeysGenerator/PHPSecLib.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\pubkey_encrypt_phpseclib\Plugin\AsymmetricKeysGenerator;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\PluginFormInterface;
+use Drupal\pubkey_encrypt\Plugin\AsymmetricKeysGeneratorBase;
+use phpseclib\Crypt\RSA;
+
+/**
+ * Provides an asymmetric keys generator based on PHPSecLib.
+ *
+ * @AsymmetricKeysGenerator(
+ *   id = "phpseclib",
+ *   name = @Translation("PHPSecLib"),
+ *   description = @Translation("RSA-based keys generated via PHPSecLib.")
+ * )
+ */
+class PHPSecLib extends AsymmetricKeysGeneratorBase implements PluginFormInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function generateAsymmetricKeys() {
+    $rsa = new RSA();
+    $keys = $rsa->createKey($this->getConfiguration()['key_size']);
+
+    // Return the keys.
+    return array(
+      "public_key" => $keys['publickey'],
+      "private_key" => $keys['privatekey'],
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function encryptWithPublicKey($original_data, $public_key) {
+    $rsa = new RSA();
+    $rsa->loadKey($public_key);
+    $encrypted = $rsa->encrypt($original_data);
+    return $encrypted;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function decryptWithPrivateKey($encrypted_data, $private_key) {
+    $rsa = new RSA();
+    $rsa->loadKey($private_key);
+    $decrypted = $rsa->decrypt($encrypted_data);
+    return $decrypted;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      'key_size' => '1024',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form['key_size'] = array(
+      '#type' => 'select',
+      '#title' => t('Key size in bits'),
+      '#options' => [
+        '1024' => '1024',
```

Comment:
I thought we weren't going to provide the small key option, but maybe I'm remembering wrong.  Can keep it in, but maybe add a '#description' here saying it's not recommended or that bigger keys are better.  We should probably add that description anyway.